tests/baselines/reference/arrayFrom.errors.txt

```diff
@@ -0,0 +1,46 @@
+tests/cases/compiler/arrayFrom.ts(19,7): error TS2322: Type 'A[]' is not assignable to type 'B[]'.
+  Type 'A' is not assignable to type 'B'.
+    Property 'b' is missing in type 'A'.
+tests/cases/compiler/arrayFrom.ts(22,7): error TS2322: Type 'A[]' is not assignable to type 'B[]'.
+
+
+==== tests/cases/compiler/arrayFrom.ts (2 errors) ====
+    // Tests fix for #20432, ensures Array.from accepts all valid inputs
+    // Also tests for #19682
+    
+    interface A {
+      a: string;
+    }
+    
+    interface B {
+      b: string;
+    }
+    
+    const inputA: A[] = [];
+    const inputB: B[] = [];
+    const inputALike: ArrayLike<A> = { length: 0 };
+    const inputARand = getEither(inputA, inputALike);
+    
+    const result1: A[] = Array.from(inputA);
+    const result2: A[] = Array.from(inputA.values());
+    const result3: B[] = Array.from(inputA.values()); // expect error

```

Comment:
This would not error out if `mapfn` was optional instead of an extra parameter in an overload